test/test_invalid_headers.py

```diff
@@ -113,6 +113,39 @@ def test_transfer_encoding_trailers_is_valid(self, frame_factory):
         request_event = events[0]
         assert request_event.headers == headers
 
+    def test_pseudo_headers_rejected_in_trailer(self, frame_factory):
+        """
+        Ensure we reject pseudo headers included in trailers
+        """
+        headers = (
+            [h for h in self.base_request_headers if h[0] != ':path'] +
+            [('trailers', 'extra-header')]
+        )
+        trailers = (
+            [(':path', '/'),
+             ('extra-header', 'values')]
+        )
+
+        c = h2.connection.H2Connection(client_side=False)
+        c.receive_data(frame_factory.preamble())
+        c.clear_outbound_data_buffer()
+
+        head_flag = ["END_HEADERS"]
+        stream_flag = ["END_STREAM"]
```

Comment:
There's no reason for us to hoist these out, they can go straight into the relevant frame factory calls.